rust/lib.rs

```diff
@@ -1,8 +1,92 @@
-#[derive(PartialEq, Debug)]
-pub struct Schema {}
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use std::fs::{read_to_string, File};
 
-pub fn get_schema(topic: &str, version: Option<u16>) -> Option<Schema> {
-    None
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[non_exhaustive]
+enum SchemaType {
+    #[serde(rename = "json")]
+    Json,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[non_exhaustive]
+enum CompatibilityMode {
+    #[serde(rename = "none")]
+    None,
+    #[serde(rename = "backward")]
+    Backward,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum SchemaError {
+    TopicNotFound,
+    InvalidVersion,
+    InvalidType,
+}
+
+impl fmt::Display for SchemaError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            SchemaError::TopicNotFound => write!(f, "Topic not found"),
+            SchemaError::InvalidVersion => write!(f, "Invalid version"),
+            SchemaError::InvalidType => write!(f, "Invalid type"),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+struct TopicSchema {
+    version: u16,
+    #[serde(rename = "type")]
+    schema_type: SchemaType,
+    compatibility_mode: CompatibilityMode,
+    resource: String,
+    examples: Vec<String>,
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+struct TopicData {
+    topic: String,
+    schemas: Vec<TopicSchema>,
+}
+
+impl TopicData {
+    fn load(path: &str) -> Result<Self, SchemaError> {
+        let f = File::open(path).map_err(|_| SchemaError::TopicNotFound)?;
+        serde_yaml::from_reader(f).map_err(|_| SchemaError::TopicNotFound)
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Schema {
+    version: u16,
+    schema_type: SchemaType,
+    compatibility_mode: CompatibilityMode,
+    // TODO: Actually parse the json schema
```

Comment:
this might be fine if we're not in the business of prescribing a specific schema validation library. otherwise I would suggest to make schema an enum like:

```
#[serde(tag = "schema_type")]
pub enum Schema {
    Json {
        schema: ...
    },
    DingusRPC {
        schema_with_some_other_type: ...
    }
}
```